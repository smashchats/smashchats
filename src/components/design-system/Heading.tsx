import React, { type PropsWithChildren } from "react";
import { DimensionValue, View, Text, TextStyle } from "react-native";

type Props = PropsWithChildren<
    {
        flex?: number;
        bg?: string;
        h?: DimensionValue;
        style?: any;
        isTruncated?: boolean;
    } & TextStyle
>;

export function Heading({
    children,
    style,
    flex,
    bg,
    h,
    fontWeight,
    marginBottom,
    isTruncated,
    color,
    fontSize,
    fontFamily,
}: Props): JSX.Element {
    return (
        <View style={{ backgroundColor: bg, height: h, flex, marginBottom }}>
            <Text
                numberOfLines={isTruncated ? 2 : undefined}
                style={{
                    color: color ?? "white",
                    fontWeight,
                    fontSize,
                    fontFamily,
                    ...style,
                }}
            >
                {children}
            </Text>
        </View>
    );
}
