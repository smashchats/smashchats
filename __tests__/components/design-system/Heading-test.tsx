import { render } from "@testing-library/react-native";
import { Heading } from "@/src/components/design-system/Heading.jsx";

describe("Heading", () => {
    test("Renders correctly", () => {
        const tree = render(
            <Heading>
                Hello
            </Heading>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
