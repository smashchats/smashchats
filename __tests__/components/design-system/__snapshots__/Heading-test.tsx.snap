// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": undefined,
      "flex": undefined,
      "height": undefined,
      "marginBottom": undefined,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontFamily": undefined,
        "fontSize": undefined,
        "fontWeight": undefined,
      }
    }
  >
    Hello
  </Text>
</View>
`;
